### PR TITLE
[COMSUP-121] Edge Functions polyfills

### PR DESCRIPTION
**Pull request by tristanlee85**

### CODE REVIEW

**Review by RaeesBhatti — DISMISSED** [review]
This is great! @tristanlee85

File: src/guides/v7/performance/cdn_as_code.md [hunk 1]
```diff
@@ -235,6 +235,37 @@ The relevant configuration options generated include the `name` and `origins` pr
...
+      },
+    ],
+
+    // Uncomment the following to configure a shield
+    // shields: { us_east: 'DCD' },
+  },
+];
+```
+
+The origin name set here will be referenced later when configuring the router to proxy requests.
```
Comment (jsernaque) [inline src/guides/v7/performance/cdn_as_code.md:267]: Perhaps move this info right below line 247:
```// The name of the backend origin
// Proxy requests to this origin by referencing this name within your router```

File: src/guides/v7/edge-functions.md [hunk 2]
```diff
@@ -27,53 +33,252 @@ export async function handleHttpRequest(request, context) {
 
 When a request is received for a route that has an Edge Function assigned to it, the Edge Function is invoked. The Edge Function is passed two parameters: `request` and `context`.
```
Comment (jsernaque) [inline src/guides/v7/edge-functions.md:34]: Consider moving the second sentence to the subsequent section.

File: src/guides/v7/edge-functions.md [hunk 3]
```diff
@@ -4,18 +4,24 @@ title: Edge Functions
 
 Edge Functions enable you to execute a small piece of JavaScript code on our edge servers. This code can be used to modify requests and responses as well as make additional calls to your defined origins. The benefits of using Edge Functions include enhancing performance by reducing latency, improving user experience by personalizing content, and increasing security by managing authentication and redirection at the edge. They allow you to execute your code closer to users, reducing the need to go back to the original server and thus, provide faster services.
 
-## Key Information {/* key_information */}
+**Key information:**
 
-Edge Functions are invoked when the `edge_function` feature is specified in your `routes.js` to assign a piece of JavaScript code a specific route:
+- Edge Functions assume you are familiar with our [CDN-as-Code](/guides/performance/cdn_as_code) approach for defining rules.
+- They require your {{ PORTAL }} team to be configured to use the Edge Functions feature. [Contact support]({{ HELP_URL }}) to enable this feature.
```
Comment (jsernaque) [inline src/guides/v7/edge-functions.md:10]:
```suggestion
- Edge Functions requires activation. {{ ACCOUNT_UPGRADE }}
```

File: src/guides/v7/edge-functions.md [hunk 4]
```diff
@@ -4,18 +4,24 @@ title: Edge Functions
...
+
+## Getting Started {/* getting-started */}
```
Comment (jsernaque) [inline src/guides/v7/edge-functions.md:12]: Remove this line. This H2 has no content.

File: src/guides/v7/edge-functions.md [hunk 5]
```diff
@@ -4,76 +4,281 @@ title: Edge Functions
...
 
-Edge Functions are invoked when the `edge_function` feature is specified in your `routes.js` to assign a piece of JavaScript code a specific route:
+- Edge Functions assume you are familiar with our [CDN-as-Code](/guides/performance/cdn_as_code) approach for defining rules.
+- Edge Functions requires activation. {{ ACCOUNT_UPGRADE }}
+
+{{ prereq.md }}
+
+## Defining Edge Functions {/* defining-edge-functions */}
+
+Edge Functions are defined in your `routes.js` file using the `edge_function` property on a route. The `edge_function` property accepts a string representing the path to the edge function file.
```
Comment (jsernaque) [inline src/guides/v7/edge-functions.md:16]:
```suggestion
Define an edge function within your {{ ROUTES_FILE }} file by adding the `edge_function` property to a route. The `edge_function` property accepts a string representing the path to the edge function file.
```

File: src/guides/v7/edge-functions.md [hunk 6]
```diff
@@ -4,76 +4,281 @@ title: Edge Functions
 
 Edge Functions enable you to execute a small piece of JavaScript code on our edge servers. This code can be used to modify requests and responses as well as make additional calls to your defined origins. The benefits of using Edge Functions include enhancing performance by reducing latency, improving user experience by personalizing content, and increasing security by managing authentication and redirection at the edge. They allow you to execute your code closer to users, reducing the need to go back to the original server and thus, provide faster services.
 
-## Key Information {/* key_information */}
+**Key information:**
 
-Edge Functions are invoked when the `edge_function` feature is specified in your `routes.js` to assign a piece of JavaScript code a specific route:
+- Edge Functions assume you are familiar with our [CDN-as-Code](/guides/performance/cdn_as_code) approach for defining rules.
```
Comment (jsernaque) [inline src/guides/v7/edge-functions.md:9]:
```suggestion
- This article assumes that you are familiar with our [CDN-as-Code](/guides/performance/cdn_as_code) approach for defining rules.
```

**Review by jsernaque — APPROVED** [review]

**Pull request merged**